kroxylicious-schema-tools/kroxylicious-schema-tool/src/main/java/io/kroxylicious/tools/schema/compiler/CodeGen.java

```diff
@@ -0,0 +1,713 @@
+import com.github.javaparser.ast.expr.InstanceOfExpr;
+import com.github.javaparser.ast.expr.MarkerAnnotationExpr;
+import com.github.javaparser.ast.expr.MemberValuePair;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.expr.Name;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.NormalAnnotationExpr;
+import com.github.javaparser.ast.expr.SimpleName;
+import com.github.javaparser.ast.expr.SingleMemberAnnotationExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.github.javaparser.ast.expr.ThisExpr;
+import com.github.javaparser.ast.expr.TypeExpr;
+import com.github.javaparser.ast.expr.TypePatternExpr;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.ast.stmt.ExpressionStmt;
+import com.github.javaparser.ast.stmt.IfStmt;
+import com.github.javaparser.ast.stmt.ReturnStmt;
+import com.github.javaparser.ast.stmt.Statement;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.type.Type;
+import com.github.javaparser.ast.type.VoidType;
+
+import io.kroxylicious.tools.schema.model.SchemaObject;
+import io.kroxylicious.tools.schema.model.SchemaObjectBuilder;
+import io.kroxylicious.tools.schema.model.SchemaType;
+import io.kroxylicious.tools.schema.model.SchemaVisitor;
+import io.kroxylicious.tools.schema.model.XKubeListType;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Java code generation from a SchemaObject.
+ */
+public class CodeGen {
+
+    private final IdVisitor idVisitor;
+    private final Diagnostics diagnostics;
+    private final Map<String, String> existingClasses;
+    private final String nullableAnnotation;
+    private final String nonNullAnnotation;
+    private final List<TypeAnnotator> typeAnnotators;
+    private final List<PropertyAnnotator> propertyAnnotators;
+
+    public CodeGen(Diagnostics diagnostics,
+                   IdVisitor idVisitor,
+                   Map<String, String> existingClasses,
+                   String nullableAnnotation,
+                   String nonNullAnnotation,
+                   List<TypeAnnotator> typeAnnotators,
+                   List<PropertyAnnotator> propertyAnnotators) {
+        this.diagnostics = Objects.requireNonNull(diagnostics);
+        this.idVisitor = Objects.requireNonNull(idVisitor);
+        this.existingClasses = existingClasses;
+        this.nullableAnnotation = nullableAnnotation;
+        this.nonNullAnnotation = nonNullAnnotation;
+        this.typeAnnotators = typeAnnotators;
+        this.propertyAnnotators = propertyAnnotators;
+    }
+
+    SchemaObject resolveRef(SchemaObject root, SchemaObject schema) {
+        var ref = schema.getRef() == null ? null : URI.create(schema.getRef());
+        if (ref != null) {
+            if (ref.isAbsolute()) {
+                diagnostics.reportFatal("Use of an absolute URI in $ref is not supported");
```

Comment:
This should be an error right? its a schema problem not a tool problem, at least thats how I've understood the classification. 

I don't think this should be a fail fast error.